swf/src/read.rs

```diff
@@ -1087,15 +1087,24 @@ impl<'a> Reader<'a> {
                 glyph.advance = Some(self.read_i16()?);
             }
 
-            for glyph in &mut glyphs {
-                glyph.bounds = Some(self.read_rectangle()?);
+            // Some older SWFs end the tag here, as this data isn't used until v7.
+            if !self.as_slice().is_empty() {
+                for glyph in &mut glyphs {
+                    glyph.bounds = Some(self.read_rectangle()?);
```

Comment:
Same here, what if a SWF has some missing glyph bounds? I guess it would respect the existing ones and leave the rest unchanged.

Reply:
If you find an example, can we make it into another PR?
This current change is kinda intuitive (the fields were added in later version, so the spec saying they are always required regardless of version is obviously wrong), your suggestions make sense it's less obvious if they should work.
For example, I've run into an swf where a "wide" font had a "non-wide" kerning-records (so trying to parse as if they were wide went out of bounds), but I couldn't verify if Flash itself accepts it, as I didn't see the font either way.

Reply:
I verified that Flash accepts SWFs where some glyph bounds are missing. In such a case I think it respects the existing ones and leaves the rest as `None`.
So it's more like:
```rs
for glyph in &mut glyphs {
    match self.read_rectangle() {
        Ok(bounds) => glyph.bounds = Some(bounds),
        Err(_) => break,
    }
}
```